public/JavaScript/gallery.js

```diff
@@ -8,18 +9,24 @@ document.addEventListener('DOMContentLoaded', () => {
     })
   }
   // Function to display selected animal
-  const showSelectedAnimals = () => {
-    hideAllAnimals(document.querySelectorAll('.gallery-items'))
-    if (selectedAnimalType) {
-      document.querySelectorAll(`.gallery-items[data-animal-type="${selectedAnimalType}"]`).forEach(item => {
-        item.style.display = 'block'
-      })
-    }
-  }
+	const showSelectedAnimals = () => {
+		hideAllAnimals(document.querySelectorAll('.gallery-items'));
+		if (selectedAnimalType) {
+			galleryItems.forEach(item => {
+				if (item.dataset.animalType === selectedAnimalType) {
+					item.style.display = 'block';
+				}
+			});
+		}
+	};
   // Target animal type
+	function setSelectedAnimalType(type) {
+		selectedAnimalType = type;
```

Comment:
This is not really encapsulation. You refactor the responsibility of changing the global state of selectedAnimalType to a single function. 
The point is to not use a global state at all. 

So instead of setting the selectedAnimalType and then using it in another function, I can just call the function which uses it and send it as parameter the selection made.